test/as_documentation/metaconstants.clj

```diff
@@ -145,22 +145,35 @@
     (provided
       --mc-- =contains=> {:c 50000})))
 
+(unfinished gen-doc)
+(fact "Test merging of metaconstant that appear in data and function fakes"
+  (:header (gen-doc)) => "gamma"
+  (provided
+    (gen-doc) => ..doc..
+    ..doc.. =contains=> {:header "gamma"}))
 
+(fact "Test merging of nested metaconstant that appear in data and function fakes"
+  (:header (:raw (gen-doc))) => "gamma"
+  (provided
+    (gen-doc) => {:raw ..doc..}
+    ..doc.. =contains=> {:header "gamma"}))
 
+(against-background [..doc.. =contains=> {:header "gamma"}]
+  (fact "Test merging of metaconstant with against-background"
+    (:header (gen-doc)) => "gamma"
+    (provided
+      (gen-doc) => ..doc..)))
 
+(against-background [..doc..    =contains=> {:header ..header..}
+                     ..header.. =contains=> {:title "title"}]
+  (future-fact "Test metaconstants that contain other metaconstants"
+    (-> (gen-doc) :header :title) => "title"
+    (provided
+      (gen-doc) => ..doc..)))
 
-
-
-
-
-
-
-
-
-
-
-;;;;
-
-;;;  Use with prerequisite functions
-
-
+(future-fact "Test metaconstants that contain other metaconstants"
+  (-> (gen-doc) :header :title) => "title"
+  (provided
+    (gen-doc) => ..doc..
+    ..doc..    =contains=> {:header ..header..}
+    ..header.. =contains=> {:title "title"}))
```

Comment:
the two `future-facts` are tests elaborating some nice-to-have behavior that currently isn't supported